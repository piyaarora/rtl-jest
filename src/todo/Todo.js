import React from 'react';

function Todo({ todo }) {
    const { id, title, completed } = todo;
    const h1 = <h3>todo - {title}</h3>
    const text = completed ? <strike>{h1}</strike> : h1;
    return <div data-testid={`todo-${id}`}>{text}</div>
}
export default Todo