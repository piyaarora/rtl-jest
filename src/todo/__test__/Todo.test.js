import { render, screen, cleanup } from '@testing-library/react'
import Todo from '../Todo'
import renderer from 'react-test-renderer'
afterEach(() => {
    cleanup();
})
test('should render non-completed todo component', () => {
    const todo = { id: 1, title: 'buy fruits', completed: false };
    render(<Todo todo={todo} />)
    const todoEl = screen.getByTestId('todo-1')
    expect(todoEl).toBeInTheDocument()
    expect(todoEl).toHaveTextContent("buy fruits")
    expect(todoEl).not.toContainHTML('<strike>')

})

test('should render completed todo component', () => {
    const todo = { id: 2, title: 'read novel', completed: true };
    render(<Todo todo={todo} />)
    const todoEl = screen.getByTestId('todo-2')
    expect(todoEl).toBeInTheDocument()
    expect(todoEl).toHaveTextContent("read novel")
})

test('matches snapshots', () => {
    const todo = { id: 1, title: 'buy fruits', completed: false };
    const tree = renderer.create(<Todo todo={todo} />).toJSON();
    expect(tree).toMatchSnapshot();
})
