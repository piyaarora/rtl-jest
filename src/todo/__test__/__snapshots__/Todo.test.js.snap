// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  data-testid="todo-1"
>
  <h3>
    todo - 
    buy fruits
  </h3>
</div>
`;
